pkg/client/client.go

```diff
@@ -240,13 +240,30 @@ func numReadyClusters(clustermesh *models.ClusterMeshStatus) int {
 	return numReady
 }
 
-// FormatStatusResponse writes a StatusResponse as a string to the writer.
-//
-// The parameters 'allAddresses', 'allControllers', 'allNodes', respectively,
-// cause all details about that aspect of the status to be printed to the
-// terminal. For each of these, if they are false then only a summary will be
-// printed, with perhaps some detail if there are errors.
-func FormatStatusResponse(w io.Writer, sr *models.StatusResponse, allAddresses, allControllers, allNodes, allRedirects, allClusters bool) {
+type StatusDetails int
+
+const (
+	// StatusAllAddress causes all addresses to be printed by FormatStatusResponse.
+	StatusAllAddresses StatusDetails = 1 << iota
+	// StatusAllControllers causes all controllers to be printed by FormatStatusResponse.
+	StatusAllControllers
+	// StatusAllNodes causes all nodes to be printed by FormatStatusResponse.
+	StatusAllNodes
+	// StatusAllRedirects causes all redirects to be printed by FormatStatusResponse.
+	StatusAllRedirects
+	// StatusAllClusters causes all clusters to be printed by FormatStatusResponse.
+	StatusAllClusters
+
+	// StatusNoDetails causes no additional details to be printed by FormatStatusResponse.
+	StatusNoDetails StatusDetails = 0
+	// StatusAllDetails causes all additional details to be printed by FormatStatusResponse.
+	StatusAllDetails StatusDetails = ^StatusDetails(0)
+)
+
+// FormatStatusResponse writes a StatusResponse as a string to the writer. The bit mask sd controls
+// whether a additional details are printed about a certain aspect of the status. In case there are
+// errors, some details may be printed regardless of the value of sd.
+func FormatStatusResponse(w io.Writer, sr *models.StatusResponse, sd StatusDetails) {
```

Comment:
Nit: have you considered using struct for StatusDetails instead?

Reply:
No haven't considered it so far, thanks for the suggestion. A `StatusDetails struct` of `bool`s would make the flag parsing in `cilium/cmd` nicer. Will change.